DataFormats/simulation/include/SimulationDataFormat/Stack.h

```diff
@@ -312,7 +311,14 @@ class Stack : public FairGenericStack
   ClassDefOverride(Stack, 1);
 };
 
-inline void Stack::addTrackReference(const o2::TrackReference& ref) { mTrackRefs->push_back(ref); }
+inline void Stack::addTrackReference(const o2::TrackReference& ref)
+{
+  if (mIndexOfCurrentTrack >= mNumberOfPrimaryParticles) {
+    Int_t iTrack = mTrackIDtoParticlesEntry[mIndexOfCurrentTrack];
+    auto& part = mParticles[iTrack];
+    part.setStore(true);
+  }
```

Comment:
it seems that the ```ref``` variable is no longer used. Probably an omission.

Reply:
indeed, thanks
corrected